bin/faucet/src/server/pow.rs

```diff
@@ -20,13 +25,16 @@ const MAX_DIFFICULTY: usize = 24;
 /// The number of active requests to increase the difficulty by 1.
 const ACTIVE_REQUESTS_TO_INCREASE_DIFFICULTY: usize = REQUESTS_QUEUE_SIZE / MAX_DIFFICULTY;
 
+/// The interval at which the challenge cache is cleaned up.
+const CLEANUP_INTERVAL_SECONDS: u64 = 5;
+
 // POW
 // ================================================================================================
 
 #[derive(Clone)]
 pub(crate) struct PoW {
     secret: [u8; 32],
-    difficulty: Arc<AtomicUsize>,
+    difficulty_per_key: Arc<Mutex<HashMap<ApiKey, usize>>>,
     challenge_cache: ChallengeCache,
```

Comment:
I would probably put `difficulty_per_key` into the `ChallengeCache` and put the mutex outside of the challenge cache - e.g.:
```rust
pub(crate) struct PoW {
    secret: [u8; 32],
    challenge_cache: Arc<Mutex<ChallengeCache>>,
}
```
This way, we'd need only one mutex.